src/Doctrine/Orm/State/LinksHandlerTrait.php

```diff
@@ -53,8 +53,10 @@ private function handleLinks(QueryBuilder $queryBuilder, array $identifiers, Que
             $hasCompositeIdentifiers = 1 < \count($identifierProperties);
 
             if (!$link->getFromProperty() && !$link->getToProperty()) {
-                $doctrineClassMetadata = $manager->getClassMetadata($link->getFromClass());
-                $currentAlias = $link->getFromClass() === $entityClass ? $alias : $queryNameGenerator->generateJoinAlias($alias);
+                // if stateOptions::entityClass is present it should be used. Otherwise it's $link::fromClass
+                $fromClass = $operation->getStateOptions()?->getEntityClass() ?? $link->getFromClass();
```

Comment:
We should fix that directly in the metadata: https://github.com/api-platform/core/blob/ccef472cade42b189fa8cb7793e27fad1349e289/src/Metadata/Resource/Factory/UriTemplateResourceMetadataCollectionFactory.php#L227

we forgot to check this option there, the condition should be something like:

https://github.com/api-platform/core/blob/main/src/Metadata/Resource/Factory/UriTemplateResourceMetadataCollectionFactory.php#L186-L188

Reply:
I wasn't sure wether that was to be fixed before the place i needed it or not. 
Aren't there possible side effects if you just replace resourceClass there? Re-checking what's actually in the Operation when i needed seemed "safer".